parsing/location.ml

```diff
@@ -423,10 +438,8 @@ let () =
       | Sys_error msg ->
           Some (errorf ~loc:(in_file !input_name)
                 "I/O error: %s" msg)
-      | Warnings.Errors n ->
-          Some
-            (errorf ~loc:(in_file !input_name)
-             "Some fatal warnings were triggered (%d occurrences)" n)
+
+      | Warnings.Errors _ -> Some non_displayed_error
```

Comment:
I think Ast_mapper.ext_of_exn would behave weirdly if I did as you say. It would rethrow the exception, and the exception would be caught at the further down, instead of being converted into an error and sent to the compiler as the other exceptions.